src/overview/actions.js

```diff
@@ -31,8 +31,25 @@ export function init() {
 // Search for docs matching the current query, update the results
 export function refreshSearch({loadingIndicator = false}) {
     return async function (dispatch, getState) {
-        const { query, startDate, endDate } = ourState(getState())
-        const oldResult = ourState(getState()).searchResult
+        let {
+            query,
+            startDate: oldStartDate,
+            endDate: oldEndDate,
+            searchResult: oldResult,
+        } = ourState(getState())
+
+        const {
+            startDate: newStartDate,
+            endDate: newEndDate,
+            extractedQuery,
+        } = extractTimeFiltersFromQuery(query)
+
+        const startDate = newStartDate || oldStartDate
+        const endDate = newEndDate || oldEndDate
+        query = extractedQuery || query
```

Comment:
As far as I understand, `extractTimeFiltersFromQuery` will always return a defined `extractedQuery`, hence this line doesn't really need to exist. `query` is just used as input to `extractTimeFiltersFromQuery`, and also means you can make the `let` on L34 a `const` as there's no need to reassign any variables. (tell me if I'm wrong) Nothing major, just something that may easily be overlooked.